server/src/controllers/courseInstance.ts

```diff
@@ -28,14 +35,31 @@ export async function getCourseInstance(req: Request, res: Response): Promise<vo
   const instanceId: number = Number(req.params.instanceId);
   await idSchema.validate({ id: instanceId });
 
-  // Find course and course instance information.
-  const course: CourseWithTranslation = await findCourseWithTranslationById(
-    courseId, HttpCode.NotFound
-  );
-
-  const instance: CourseInstance | null = await CourseInstance.findByPk(
-    instanceId
-  );
+  const instance: CourseInstanceWithCourseAndTranslation | null =
+    await models.CourseInstance.findByPk(
+      instanceId,
+      {
+        include: [
+          {
+            model: Course,
+            attributes: ['id', 'courseCode'],
+            include: [
+              {
+                model: CourseTranslation,
+                attributes: ['language', 'courseName', 'department']
+              }
+            ]
+          },
+          {
+            model: User,
+            attributes: ['name'],
+            where: {
+              '$Users->CourseInstanceRole.role$': CourseInstanceRoleType.TeacherInCharge
```

Comment:
Just checking, does Sequelize guarantee that this dollar-sign notation is valid?

Reply:
Documentation for (part of?) this can be found here: https://sequelize.org/docs/v6/advanced-association-concepts/eager-loading/#complex-where-clauses-at-the-top-level

Does this answer your question? If not, could you elaborate on what you mean?

Reply:
That explains it! I was wondering if this was an undocumented hack at first, but it's all good :+1: